From 2ac210b2b683ba213aa75d771d7408be21c11ec5 Mon Sep 17 00:00:00 2001
From: Raymond Wong <infiwang@pm.me>
Date: Fri, 24 Feb 2023 17:12:01 +0800
Subject: [PATCH 062/156] riscv(asm): fix asm_tointg guard

Refering to other arch, the guard seems to trigger on not equal.

This is a WIP, archive only.
---
 src/lj_asm_riscv64.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lj_asm_riscv64.h b/src/lj_asm_riscv64.h
index df277dc7..1e0171c9 100644
--- a/src/lj_asm_riscv64.h
+++ b/src/lj_asm_riscv64.h
@@ -440,7 +440,7 @@ static void asm_tointg(ASMState *as, IRIns *ir, Reg left)
 {
   Reg tmp = ra_scratch(as, rset_exclude(RSET_FPR, left));
   Reg dest = ra_dest(as, ir, RSET_GPR), cmp = ra_scratch(as, RSET_GPR);
-  asm_guard(as, RISCVI_BNE, cmp, RID_ZERO);
+  asm_guard(as, RISCVI_BEQ, cmp, RID_ZERO);
   emit_ds1s2(as, RISCVI_FEQ_D, cmp, tmp, left);
   emit_ds(as, RISCVI_FCVT_D_W, tmp, dest);
   emit_ds(as, RISCVI_FCVT_W_D, dest, left);
-- 
2.42.0

